gcovr/writer/html.py

```diff
@@ -482,7 +484,7 @@ def branch_coverage_class(percent: Optional[float]) -> str:
                     )
         except IOError as e:
             logger.warning(f'File {data["filename"]} not found: {repr(e)}')
-            for ctr in range(1, max_line_from_cdata):
+            for ctr in range(1, max_line_from_cdata + 1):
```

Comment:
Why using `+ 1` here?

Reply:
Well spotted. I should have added a comment, but I did explain it in the commit message <https://github.com/gcovr/gcovr/pull/663/commits/26b1dea1a85b1294a723f50671a3e6f883e4943d>:

>  html writer: fix handling of nonexistent and empty files
>
> Previously, one to few dummy lines were inserted for nonexistent files

Earlier, we have `max_line_from_cdata = max(cdata.lines.keys(), default=0)`. That is, this variable contains the maximum line number that exists in the coverage data.

Python ranges are exclusive. We want to iterate over all lines, *including* that last line. Thus, we have to add a `+1` to include that line.

Alternatively, we could have looped over `range(max_line_from_cdata)` and then added `+1` within the loop body since line numbers use 1-based indexing.